secrets/awskms/kms.go

```diff
@@ -96,12 +97,13 @@ func (o *lazySessionOpener) OpenKeeperURL(ctx context.Context, u *url.URL) (*sec
 // Scheme is the URL scheme awskms registers its URLOpener under on secrets.DefaultMux.
 const Scheme = "awskms"
 
-// URLOpener opens AWS KMS URLs like "awskms://keyID".
+// URLOpener opens AWS KMS URLs like "awskms://keyID" or "awskms:///keyID".
 //
 // The URL Host + Path are used as the key ID, which can be in the form of an
 // Amazon Resource Name (ARN), alias name, or alias ARN. See
 // https://docs.aws.amazon.com/kms/latest/developerguide/viewing-keys.html#find-cmk-id-arn
-// for more details.
+// for more details. Note that ARNs may contain ":" characters, which cannot be
```

Comment:
This may need more clarity in what the exact case we're handling? The issue reporting this had two //s before the ARN

Reply:
Before this PR, we concatenated the Host and Path to form the KeyID. ARNs did not work correctly because they have ":" characters, which can't be escaped in the Host, and the parsing expects to be followed by a port number. So, the URLs in the bug report don't parse.

You could always leave the Host empty (e.g., "awskms:///foo:bar"), even before this PR, but since Paths start with a "/", you'd end up with a KeyID like "/foo:bar", which wouldn't work.

In this PR, I'm checking for that case and dropping the leading "/" if the Host is empty. So the user who filed the bug will have to change their URLs to have "awskms:///" instead of "awskms://" and then it will work.

This docstring is pointing out that if your ARN has ":", you need to use the 3-slash form.